### DISCUSSION TITLE
Mock/Simulate OBS/SLOBS and RockSniffer "connection", responses.

### DISCUSSION THREAD

**Discussion by kozaka-tv** [Th0rstenf/Rocksniffer-to-Streamer.bot-extension#111] [Ideas]
It would be great, if we in our IDE run the app, that we could use and test the function without live OBS and Sniffer connection. Either Mocked and/or live connected.

1) Introduce in Mock a connection to OBS/SLOBS (in config configurable websocket connection)
2) Maybe there is a better solution, but what if we have a file, where we set the actual Sniffer and Stream App values for the Getters?

-------
To 1) If we have a live connection, then we do not need to copy every single change into SB to test, rather than just at the end to be sure, it is really working (like because version and system differences it could be that in IDE it works, but in SB not).

To 2) If we could mock the answers/states of Sniffer and the Stream Apps, then we could just run the app in the IDE, and then simulate the behaviors of all the apps, and test/debug in IDE. Like, have a file, where we could set the values manually, and this will be used for all the Get/Set functions. This way, I do not have to run my 2 PC-s, just one, and we do not need to start apps like RS, what is consuming a lot of energy for nothing. But the point is, to have a more flexible testing/debugging environment. This will be also handy for the Tests....maybe only the test would be also enough. Or we could have a small GUI where I could klick scenes for OBS, or set game state for Sniffer mocks. Read the App actions.

---
https://github.com/BarRaider/obs-websocket-dotnet
https://www.nuget.org/packages/obs-websocket-dotnet/4.9.1

**Comment by Th0rstenf** [comment]
Our interface is CPH. The whole reason for integrating it into SB was to **get away** from needing to maintain the OBS websockets.
In fact there is not a single direct connection to OBS/SLOBS
For debugging it is sufficient to check whether the corresponding CPH methods are called. It really doesn't add any real value, just adds tons of effort

**Reply by kozaka-tv** [reply]
Yes, maybe to add here the Websocket stuff was not the best, as really, we do not have an OBS/SLOBS connection at all.

What I ment with this, is that if there would be some solution to simulate an OBS, that would be great. Like the methods ObsSetScene() and ObsGetCurrentScene() (+SLOBS) are changed by OBS and if that would be not the case, rather than that would be a "feature" in CPHMock or something, then we could just debug/test everything in the IDE. 

Maybe it sounds like it is ton of a work, but if I have time, and we do not have too much to do, I would at least write some Tests. And for the testing that would be also probably useful.

To test something, right now, I need to run 2 PC-s, and that consumes a lot of energy. If that would be a text file, or something, a small GUI, would be way more better.

Shit! Also Sniffer should be simulated...damn.

**Reply by Th0rstenf** [reply]
Properly simulating CPH and sniffer could be more work than the project itself, i'd prefer having unit tests that simulate the record input (leaving the sniffer and parser out of scope, at least initially) and checking if expected calls to CPH were made.

Even that is still  a lot of effort, but a fraction of completely simulating the outside world. I wanted to integrate in SB to reduce maintenance after all
